src/config/agentlessd-config.c

```diff
@@ -146,15 +205,63 @@ int Read_CAgentless(XML_NODE node, void *config, __attribute__((unused)) void *c
     if (!lessd_config->entries[s]->server ||
             !lessd_config->entries[s]->state ||
             !lessd_config->entries[s]->type) {
-        merror(XML_INV_MISSOPTS);
+        if (output == NULL){
+            merror(XML_INV_MISSOPTS);
+        } else {
+            wm_strcat(output, "Missing agentless options.", '\n');
+        }
         return (OS_INVALID);
     }
 
     if ((lessd_config->entries[s]->state == LESSD_STATE_PERIODIC) &&
             !lessd_config->entries[s]->frequency) {
-        merror(XML_INV_MISSFREQ);
+        if (output == NULL) {
+            merror(XML_INV_MISSFREQ);
+        } else {
+            wm_strcat(output, "Frequency not set for the periodic option.", '\n');
+        }
         return (OS_INVALID);
     }
 
     return (0);
 }
+
+int Test_Agentlessd(const char *path, char **output) {
+    agentlessd_config *agtless_config;
+    os_calloc(1, sizeof(agentlessd_config), agtless_config);
+
+    if(ReadConfig(CAGENTLESS, path, agtless_config, NULL, output) < 0) {
+        if (output == NULL){
+            merror(XML_INV_AGENTLESS);
+        } else {
+            wm_strcat(output, "ERROR: Invalid configuration in Agentless", '\n');
+        }
+        free_AgentlessConfig(agtless_config);
+        return OS_INVALID;
+    }
+
+    // Free Memory
+    free_AgentlessConfig(agtless_config);
+    return 0;
+}
+
+void free_AgentlessConfig(agentlessd_config *config) {
+    if(config) {
+        int i, j;
```

Comment:
The scope of the variables `i` and `j` should be inside the conditional `if(config->entries)`